backend/src/app.ts

```diff
@@ -1,28 +1,47 @@
-import "dotenv/config";
-import express, { Express } from "express";
-import mongoose from "mongoose";
-import cors from "cors";
-import bodyParser from "body-parser";
-import authRouter from "./routes/authRouter";
+import express from 'express'
+import cors from 'cors'
+import bodyParser from 'body-parser'
+import authRouter from './routes/auth.router'
+import { connectDatalake, connectMetadata } from './db'
+import swaggerJsDoc from 'swagger-jsdoc'
+import swaggerUi from 'swagger-ui-express'
 
-const app = express();
+const app = express()
 
-const initApp = () => {
-  app.use(cors());
-  app.use(bodyParser.json());
-  app.use(bodyParser.urlencoded({ extended: true }));
-  app.use("/auth", authRouter);
+const initApp = async () => {
+  app.use(cors())
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({ extended: true }))
+  app.use('/auth', authRouter)
 
-  return new Promise<Express>((resolve, reject) => {
-    mongoose
-      .connect(process.env.DB_URL)
-      .then(() => {
-        resolve(app);
-      })
-      .catch((err) => {
-        reject(err);
-      });
-  });
-};
+  await connectDatalake()
+  await connectMetadata()
```

Comment:
If you want you can initiate both async using `Promise.all` and calling on both connections